src/api/institutionApi.ts

```diff
@@ -24,3 +27,47 @@ export const getInstitutionAndSubunits = async (searchTerm: string) => {
   // }
   return mockInstitutionResponse;
 };
+
+export const getParentUnits = async (subunitid: string) => {
+  // TODO: get institutions from endpoint
+  // BACKEND NOT FINISHED YET
+  try {
+    const idToken = await getIdToken();
+    const headers = {
+      Authorization: `Bearer ${idToken}`,
+    };
+    const url = `${InstituionApiPaths.INSTITUTION}?id=${subunitid}`;
+    const response = await Axios.get(url, { headers });
+    if (response.status === StatusCode.OK) {
+      const { id, name, subunits } = response.data;
+
+      if (subunits.length > 0) {
+        let unit: RecursiveUnit = { name: '', id: '', subunits: [] };
+        unit.id = id;
+        unit.name = name;
+        unit.subunits = getSubunits(subunits);
+        return unit;
+      } else {
+        return response.data;
+      }
+    } else {
+      return null;
+    }
+  } catch {
+    return { error: i18n.t('feedback:error.get_parent_units') };
+  }
+};
+
+// inspired by https://stackoverflow.com/questions/48171842/how-to-write-a-recursive-flat-map-in-javascript
```

Comment:
jiiiz, folk har blitt kastet på bålet for mindre heksekunst enn dette :D men gjør den jobben så gjør den jobben ;)

Reply:
i know. men dette klarte jeg ikke å tenke meg til selv, så måtte ty til hjelp 😅